crates/project/src/worktree.rs

```diff
@@ -1301,9 +1310,29 @@ impl LocalWorktree {
         let abs_old_path = self.absolutize(&old_path);
         let abs_new_path = self.absolutize(&new_path);
         let fs = self.fs.clone();
+        let case_sensitive = self.fs_case_sensitive;
         let rename = cx.background_executor().spawn(async move {
-            fs.rename(&abs_old_path?, &abs_new_path?, Default::default())
-                .await
+            let abs_old_path = abs_old_path?;
+            let abs_new_path = abs_new_path?;
+
+            let abs_old_path_lower = abs_old_path.clone().to_str().map(|p| p.to_lowercase());
+            let abs_new_path_lower = abs_old_path.clone().to_str().map(|p| p.to_lowercase());
```

Comment:
I believe we should use the new path here instead:

```suggestion
            let abs_new_path_lower = abs_new_path.to_str().map(|p| p.to_lowercase());
```